app/util/analytics.py

```diff
@@ -114,24 +125,63 @@ def get_events(since_timestamp=None, since_id=None):
     since_timestamp = float(since_timestamp) if since_timestamp else 0.0
     since_id = int(since_id) if since_id else 0
 
-    with open(_eventlog_file, 'r') as f:
-        # todo(joey): This is inefficient since it reads the whole log file into memory (can be several megabytes).
-        # We probably want something like http://code.activestate.com/recipes/120686/
-        reversed_log_lines = reversed(f.readlines())
+    filtered_events_from_cache = get_events_from_cache(since_timestamp, since_id)
+    if len(filtered_events_from_cache) > 0:
```

Comment:
Oh yeah that's what I meant -- the user might specify a timestamp from a long time ago, or a since_id of 0 (or just specify nothing at all) which may push the number of events we'd like to return over that `LOG_CACHE_MAX_SIZE` limit. 

It seems like in that case we'll just return the entire cache and then stop, even though it may be appropriate to go load even more events from the file.

Actually it seems like currently we would never load any events from the file at all since the cache would always win.